base_phone_popup/models/phone_common.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2014-2016 Akretion, Alexis de Lattre <alexis.delattre@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, models
+import logging
+
+
+_logger = logging.getLogger(__name__)
+
+
+class PhoneCommon(models.AbstractModel):
+    _inherit = 'phone.validation.mixin'
+
+    @api.model
+    def get_record_from_phone_number(self, presented_number):
+        """If it finds something, it returns (object name, ID, record name)
+        For example : ('res.partner', 42, u'Alexis de Lattre (Akretion)')
+        """
+        _logger.debug(
+            "Call get_name_from_phone_number with number = %s"
+            % presented_number)
+        if not presented_number.isdigit():
+            _logger.warning(
+                "Number '%s' should only contain digits." % presented_number)
+        nr_digits_to_match_from_end = \
```

Comment:
I'm not sure if this should be at company level, or system wide (through a system parameter). What can be the reasoning for having different lengths for company?